### PR TITLE
first submission, needs a graphic

### PR DESCRIPTION
**Pull request by bolangi**
Hiya thank you. Here is my article. Hope you can find a day for it.
I will send a suitable graphic shortly. How do you mark up for
an included image?

### CODE REVIEW

**Comment by bolangi** [comment]
This pull request resolves (Issue #301) 

Thank you, and grateful for any input.

**Comment by oalders** [comment]
For images, you can included arbitrary HTML in the article:

```pod
=for html
<img src="12-05-naughty.northpole.io.png" />

```

Images themselves can live in `2023/share/static`.

**Comment by oalders** [comment]
I guess this still needs the graphic?

File: 2023/incoming/trimming-audio-files-with-audio-nama.pod [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+ L<Role::Tiny>: Breaking up large classes
+ L<Term::ReadLine::Gnu>: Terminal support, tab completion, history
+ L<Term::TermKey>: Hotkey mappings
+ L<Graph>: Representing, traversing and transforming the signal processing network
+ L<AnyEvent>: Event loop
+ L<Tk>: GUI, available with the -g option, use at your own risk!
+ L<Data::Section::Simple>: Store and access sections within the __DATA__ block
+ L<Data::Dumper::Concise>, tho I would now recommend L<Data::Printer>.
+ L<Memoize>: Turn expensive, redundant calculations into cheap lookups
+ L<Devel::NYTProf>: Profiling to find and solve bottlenecks
```
Comment (openstrike) [inline 2023/incoming/trimming-audio-files-with-audio-nama.pod:338]: This appears to be a list. Perhaps it should start with an =over, then be a sequence of =item nodes and finally a =back?

**Review by openstrike — CHANGES_REQUESTED** [review]
While this is all very interesting, I'm left wondering how much it has to do with Perl. There is a reference to Audio::Nama (which itself appears to have no POD) but that's about it. It would be good to have a little more of the Perl side of things, ideally. Even if that's only to re-enforce that Audio::Nama is a Perl dist.

**Comment by bolangi** [comment]
Hey Pete

Thank you for reponding,

Audio::Nama is a perl distribution that provides /usr/bin/nama,
an audio processing application. 

Pardon me for the many words as I stand up for 
my article and module!

I have revised the article, separating the actions from the
technical details. I am satisfied with it, think it flows
well and represents a significant offering.

To me, the list of dependencies shows how a few great
behaviors from perl modules can combine to something
unexpected or large in scope.

You write are concerned that there aren't code examples using
Audio::Nama. Well, yes it is an application. 

And true, most of the code examples are not perl code, but
they are code our perl program has generated.  I considered
that the topic of generating language is evergreen. Parsing,
too. 

We're exposing a new command language primarily via
the terminal. We are concisely eliciting behaviors as we
would in a perl script. 

Personally, I think it is amazing how much is accomplished
from a few words. 

I think it's a good demonstration of something that could be
considered a large task (audio workstation) brought to
fruition with a modicum of perl and, if I may say, some
tenacity, by targeting an excellent, existing library.

I will send you a new/updated pull request with my much revised
text, including a graphic, and fixing the pod.

Please have a look at upcoming request and judge if it acceptable.


-- 
Joel Roth

**Comment by bolangi** [comment]
I have the complete article with updates that address maintainer's  issues.  How shall I submit it? A new pull request?

**Comment by bolangi** [comment]
It surprises me you got my updates without my sending them explicitly.

**Comment by bolangi** [comment]
I will be uploading a new version to CPAN, will do my best to make sure this is done before the article is published, which you suggested could be on the 17th or 18th.

**Comment by bolangi** [comment]
I've got the upload done.

Thank you.

**Review by openstrike — APPROVED** [review]
Thanks, @bolangi! This most recent version highlights the use of Perl underpinning the nama application nicely.

The article looks ready to me.

**Comment by oalders** [comment]
This has now been scheduled for Dec 18. I like the use of the small graphic as a divider. 😄 Thanks @bolangi and @openstrike!

**Comment by bolangi** [comment]
Please advise how it will be possible to submit 
a couple small, signicant changes.


-- 
Joel Roth

**Comment by bolangi** [comment]
I'm sure you don't touch them anymore, but FWIW, here's the
diff 

diff --git a/2023/incoming/trimming-audio-files-with-audio-nama.pod b/2023/incoming/trimming-audio-files-with-audio-nama.pod
index 94206f7..d4c907c 100644
--- a/2023/incoming/trimming-audio-files-with-audio-nama.pod
+++ b/2023/incoming/trimming-audio-files-with-audio-nama.pod
@@ -30,7 +30,7 @@ help.
 
 Nama configures Ecasound for recording, mixing and a variety
 of tasks via a audio network definition called a I<chain
-setup> with syntax we will use Nama to generate. 
+setup> with syntax we will use perl to generate. 
 
 Aside from housekeeping duties, Nama's main task is to
 regenerate the chain setup (and re-configure the engine)
@@ -214,7 +214,7 @@ part of an audio file that plays at a particular time.
        -a:5 -f:s16_le,2,44100 -i:playat,40,select,80,300,/home/jroth/nama/christmas/.wav/carol_1.wav
 
 The 'select,80,300' object extracts 300 seconds of audio
-starting at 0:80. The 'playat,40' option plays the audio
+starting at 0:80. The 'playat,40' object plays the audio
 after a delay of 40 seconds.
-- 
Joel Roth

**Comment by openstrike** [comment]
> Please advise how it will be possible to submit a couple small, signicant changes.

1. Ensure your fork is up-to-date with the main branch here.
2. Create a branch on your fork for the changes.
3. Modify your article (which is now [2023-12-18.pod](https://github.com/perladvent/Perl-Advent/blob/main/2023/articles/2023-12-18.pod) as desired and then submit a PR.

Thanks.

**Pull request merged**